internal/queue/pending_txs_test.go

```diff
@@ -0,0 +1,75 @@
+// Copyright IBM Corp. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package queue
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/IBM-Blockchain/bcdb-server/pkg/types"
+	"github.com/golang/protobuf/proto"
+	"github.com/stretchr/testify/require"
+)
+
+func TestPendingTxs_Async(t *testing.T) {
+	pendingTxs := NewPendingTxs()
+
+	var p *completionPromise
+	require.True(t, pendingTxs.Empty())
+	pendingTxs.Add("tx1", p)
+	require.True(t, pendingTxs.Has("tx1"))
+	require.False(t, pendingTxs.Has("tx2"))
+	pendingTxs.Add("tx2", p)
+	require.True(t, pendingTxs.Has("tx2"))
+	pendingTxs.DoneWithReceipt([]string{"tx1", "tx2"}, nil)
+	require.True(t, pendingTxs.Empty())
+}
+
+func TestPendingTxs_Sync(t *testing.T) {
+	pendingTxs := NewPendingTxs()
+	p := NewCompletionPromise(time.Hour)
+	pendingTxs.Add("tx3", p)
+
+	blockHeader := &types.BlockHeader{
+		BaseHeader: &types.BlockHeaderBase{
+			Number:                5,
+			LastCommittedBlockNum: 1,
+		},
+	}
+	expectedReceipt := &types.TxReceipt{
+		Header:  blockHeader,
+		TxIndex: 0,
+	}
+	var wg sync.WaitGroup
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		actualReceipt, err := p.Wait()
+		require.NoError(t, err)
+		require.True(t, proto.Equal(expectedReceipt, actualReceipt))
+	}()
+
+	pendingTxs.DoneWithReceipt([]string{"tx3"}, blockHeader)
+	wg.Wait()
```

Comment:
This is fine but we can avoid the WaitGroup by placing the `DoneWithReceipt()` within the groutine (with a sleep if needed) and place the p.Wait() outside as it is waiting for the `DoneWithReceipt()` anyway.

Reply:
Done